ui/StatusQ/src/shellproxymodel.cpp

```diff
@@ -0,0 +1,234 @@
+#include "StatusQ/shellproxymodel.h"
```

Comment:
I think that we should add generic model to our toolbox instead of creating case-specific model there. It could be kind of "lightweight" version of WritableProxyModel, where we can define additional writable roles.

There is a chance it can be done on top of existing internal utility `ModelSyncedContainer` allowing to store some additional metadata synced with the model but with O(1) read/write.

Reply:
Another idea is to use one more `ObjectProxyModel` for now, in very simple form:

```qml
    ObjectProxyModel {
        delegate: QtObject {
            property real timestamp: 0
            property bool pinned: false
        }

        exposedRoles: ["timestamp", "pinned"]

        sourceModel: srcModel
    }
```
Basically it adds two additional writable roles. It should work nicely with separated loading/saving.

Reply:
OK, did it as you said :) Custom C++ model dropped completely